### ISSUE TITLE
remove hard dependency on wdio-sync or fibers

### ISSUE THREAD

**Issue by jedwards1211** [webdriverio/wdio-mocha-framework#26]
I just use async functions in my `describe` and `it` and have no use for `wdio-sync`.  I would prefer not to have `fibers` lurking around in my subdeps.  It's not 100% necessary for using mocha, is it?

Alternatively I'm thinking of looking into making a `wdio-async-mocha-framework` (and a corresponding `wdio-async` with the equivalent `wrapCommands` etc.)

**Comment by christian-bromann** [comment]
@jedwards1211 wdio-sync also makes sure that it runs all hooks in your wdio.conf.js, which is essential to run all services. Sure you can separate that logic somehow but why is having `fibers` around such a problem? I rather would keep it around to prevent confusions for the majority of people using fibers.

**Comment by christian-bromann** [comment]
> Alternatively I'm thinking of looking into making a wdio-async-mocha-framework

That is a great alternative.

Why are you using wdio in asynchronous mode? Things are so much easier running everything synchronous (just curious).

**Comment by jedwards1211** [comment]
I have several reasons:
* It's easier to do things in parallel with `async`/`await`, like read the text of a bunch of components
* Fibers will always lag behind Node releases when the V8 API changes.
* I like the explicitness of `async`/`await`
I'm thinking I should make a `wdio-async` package that runs all the hooks as well (`wdio-sync` ironically does seem to have an async mode and seems to handle promises returned from my hooks just fine)

**Comment by christian-bromann** [comment]
> like read the text of a bunch of components at once

This happens also in sync mode synchronously: if you call getText on a selector that matches multiple elements, all elementIdText calls are made in parallel

> wdio-sync ironically does seem to have an async mode and seems to handle promises returned from my hooks just fine

I agree. wdio-sync might not be the best name for its purpose anymore.

I think you will be fine creating a wdio-async package that has no Fibers dependency and remove all future calls. Then just use shrinkwrap to replace it with wdio-sync.

**Comment by jedwards1211** [comment]
> This happens also in sync mode synchronously: if you call getText on a selector that matches multiple elements, all elementIdText calls are made in parallel

I'm talking about multiple distinct selectors at once:
```es6
await Promise.all(['firstName', 'lastName'].map(async field => {
  expect(await browser.getText(`#${field}`).to.equal(expectedValues[field])
})
```
Not really a game changer, but a nice to have in my opinion.

Yeah, I understand it would confuse users.  Though I think as async/await becomes more widely used, Fibers will become less well known and seem kind of baroque to new users.

**Comment by christian-bromann** [comment]
> Though I think as async/await becomes more widely used, Fibers will become less well known and seem kind of baroque to new users.

Probably. Let's see. Still, many QA (non devs) and people who switch over from Java to Node.JS would have a hard time figuring this out.

**Comment by jedwards1211** [comment]
@christian-bromann yeah...though Java dropped the ball on that one.  C# has had async/await for a long time

**Comment by jedwards1211** [comment]
@christian-bromann if `isAsync()` is true, then fibers is not used anyway right?  Meaning you could conditionally `require('fibers')` instead of always importing it, and put `fibers` in `optionalDependencies`.

**Comment by christian-bromann** [comment]
> That would be a very simple solution to this issue.

Probably for you. But I already can see lots of people who are curious why they get an error message due to a missing dependency.

**Comment by timruffles** [comment]
Please could you throw an error when this is used with `sync: false`? It's rather frustrating when things silently fail.

**Comment by christian-bromann** [comment]
> It's rather frustrating when things silently fail.

How does it fail silently?
